dialogue-core/src/main/java/com/palantir/dialogue/core/DialogueChannel.java

```diff
@@ -129,14 +130,7 @@ public void updateUris(Collection<String> uris) {
                 .build();
         newUris.forEach(uri -> limitedChannelByUri.put(uri, createLimitedChannel(uri, allUris.indexOf(uri))));
 
-        nodeSelectionStrategy.getAndUpdate(previous -> getUpdatedNodeSelectionStrategy(
-                previous,
-                clientConfiguration,
-                ImmutableList.copyOf(limitedChannelByUri.values()),
-                random,
-                ticker,
-                channelName));
-
+        nodeSelectionStrategy.updateChannels(ImmutableList.copyOf(limitedChannelByUri.values()));
```

Comment:
We could keep things simpler by putting our complete trust in the most recent response we've received, that way the nodeSelectionStrategy wouldn't need to wrap individual uri channels or be notified of uri updates.

No real proposal here, just weighing implementing this in a more naive way to begin with.

Reply:
Ya I had thought of that originally, but it would lead to pretty bad behaviour during rolling upgrades with RR/Balanced. I think the extra bit of complexity is worth handling the case